### ISSUE TITLE
Connection Issue with Providers

### ISSUE THREAD

**Issue by krewsayder** [vn-ki/anime-downloader#100]
**Describe the bug**
Issue using anime-downloader with connections.  Issue appears on 9anime and kissanime.  I've put the DEBUGS below.  I'm wondering if this is some type of proxy issue, but unclear.

I've gotten the issue to not appear on other providers, however those providers return a 'not found' error.  The "not found" errors appear after trying 720p, 480p, etc.

**To reproduce**

```

```

<details>
<summary>LOG</summary>

```

C:\Users\Hoogstra>anime dl "Tokyo Ghoul" --provider kissanime --download-dir H:\AnimeDownload -q 1080p -ll DEBUG
INFO root: anime-downloader 3.6.0
DEBUG root: Platform: Windows-10-10.0.17134-SP0
DEBUG root: Python 3.6.4
DEBUG root: Tokyo Ghoul (Dub)
DEBUG root: Tokyo Ghoul (Sub)
DEBUG root: Tokyo Ghoul 2 (Dub)
DEBUG root: Tokyo Ghoul 2 (Sub)
DEBUG root: Tokyo Ghoul 3 (Dub)
DEBUG root: Tokyo Ghoul 3 (Sub)
DEBUG root: Tokyo Ghoul 4 (Dub)
DEBUG root: Tokyo Ghoul 4 (Sub)
DEBUG root: Tokyo Ghoul: "Jack"
DEBUG root: Tokyo Ghoul: "Pinto"
 1: Tokyo Ghoul (Dub)
 2: Tokyo Ghoul (Sub)
 3: Tokyo Ghoul 2 (Dub)
 4: Tokyo Ghoul 2 (Sub)
 5: Tokyo Ghoul 3 (Dub)
 6: Tokyo Ghoul 3 (Sub)
 7: Tokyo Ghoul 4 (Dub)
 8: Tokyo Ghoul 4 (Sub)
 9: Tokyo Ghoul: "Jack"
10: Tokyo Ghoul: "Pinto"

Enter the anime no:  [1]: 2
INFO root: Selected Tokyo Ghoul (Sub)
INFO root: Extracting episode info from page
DEBUG root: Unfiltered episodes : ['http://kissanime.ru/Anime/Tokyo-Ghoul/_EX02?id=97946', 'http://kissanime.ru/Anime/Tokyo-Ghoul/_EX01?id=97945', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-012?id=97944', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-011?id=97943', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-010?id=97942', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-009?id=97941', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-008?id=97940', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-007?id=97939', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-006?id=97938', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-005?id=97937', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-004?id=97936', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-003?id=97935', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-002?id=97934', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-001?id=97933']
DEBUG root: Filtered episodes : ['http://kissanime.ru/Anime/Tokyo-Ghoul/_EX02?id=97946', 'http://kissanime.ru/Anime/Tokyo-Ghoul/_EX01?id=97945', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-012?id=97944', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-011?id=97943', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-010?id=97942', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-009?id=97941', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-008?id=97940', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-007?id=97939', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-006?id=97938', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-005?id=97937', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-004?id=97936', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-003?id=97935', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-002?id=97934', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-001?id=97933']
DEBUG root: EPISODE IDS: length: 14, ids: ['http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-001?id=97933', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-002?id=97934', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-003?id=97935', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-004?id=97936', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-005?id=97937', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-006?id=97938', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-007?id=97939', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-008?id=97940', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-009?id=97941', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-010?id=97942', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-011?id=97943', 'http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-012?id=97944', 'http://kissanime.ru/Anime/Tokyo-Ghoul/_EX01?id=97945', 'http://kissanime.ru/Anime/Tokyo-Ghoul/_EX02?id=97946']
INFO root: Found anime: Tokyo Ghoul (Sub)
INFO root: Downloading to H:\AnimeDownload
DEBUG root: Extracting stream info of id: http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-001?id=97933
DEBUG root: Calling url: http://kissanime.ru/Anime/Tokyo-Ghoul/Episode-001?id=97933&s=rapidvideo
DEBUG root: Sources :
DEBUG root: Calling Rapid url: https://www.rapidvideo.com/e/FF2UPWJL6H&q=1080p
INFO root: Downloading Tokyo Ghoul (Sub)-1
INFO root: H:\AnimeDownload\Tokyo_Ghoul_Sub/Tokyo_Ghoul_Sub_1.mp4
Traceback (most recent call last):
  File "c:\programdata\anaconda3\lib\site-packages\urllib3\connection.py", line 159, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw)
  File "c:\programdata\anaconda3\lib\site-packages\urllib3\util\connection.py", line 57, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "c:\programdata\anaconda3\lib\socket.py", line 745, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 11001] getaddrinfo failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\programdata\anaconda3\lib\site-packages\urllib3\connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "c:\programdata\anaconda3\lib\site-packages\urllib3\connectionpool.py", line 343, in _make_request
    self._validate_conn(conn)
  File "c:\programdata\anaconda3\lib\site-packages\urllib3\connectionpool.py", line 839, in _validate_conn
    conn.connect()
  File "c:\programdata\anaconda3\lib\site-packages\urllib3\connection.py", line 301, in connect
    conn = self._new_conn()
  File "c:\programdata\anaconda3\lib\site-packages\urllib3\connection.py", line 168, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
urllib3.exceptions.NewConnectionError: <urllib3.connection.VerifiedHTTPSConnection object at 0x0000016A08D2EC50>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\programdata\anaconda3\lib\site-packages\requests\adapters.py", line 449, in send
    timeout=timeout
  File "c:\programdata\anaconda3\lib\site-packages\urllib3\connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "c:\programdata\anaconda3\lib\site-packages\urllib3\util\retry.py", line 398, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='www678.playercdn.net', port=443): Max retries exceeded with url: /173/0/ZTUNk-BVQjUZwYQK-4tVSg/1540713258/170427/NLF1Av6zZTQd4H7.mp4 (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000016A08D2EC50>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\programdata\anaconda3\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\programdata\anaconda3\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\ProgramData\Anaconda3\Scripts\anime.exe\__main__.py", line 9, in <module>
  File "c:\programdata\anaconda3\lib\site-packages\click\core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "c:\programdata\anaconda3\lib\site-packages\click\core.py", line 717, in main
    rv = self.invoke(ctx)
  File "c:\programdata\anaconda3\lib\site-packages\click\core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\programdata\anaconda3\lib\site-packages\click\core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\programdata\anaconda3\lib\site-packages\click\core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "c:\programdata\anaconda3\lib\site-packages\click\decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "c:\programdata\anaconda3\lib\site-packages\anime_downloader\cli.py", line 146, in dl
    range_size=chunk_size)
  File "c:\programdata\anaconda3\lib\site-packages\anime_downloader\sites\anime.py", line 183, in download
    path, force, range_size=range_size)
  File "c:\programdata\anaconda3\lib\site-packages\anime_downloader\downloader\base_downloader.py", line 23, in __init__
    r = session.get_session().get(self.url, stream=True)
  File "c:\programdata\anaconda3\lib\site-packages\requests\sessions.py", line 537, in get
    return self.request('GET', url, **kwargs)
  File "c:\programdata\anaconda3\lib\site-packages\requests\sessions.py", line 524, in request
    resp = self.send(prep, **send_kwargs)
  File "c:\programdata\anaconda3\lib\site-packages\requests\sessions.py", line 637, in send
    r = adapter.send(request, **kwargs)
  File "c:\programdata\anaconda3\lib\site-packages\requests\adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='www678.playercdn.net', port=443): Max retries exceeded with url: /173/0/ZTUNk-BVQjUZwYQK-4tVSg/1540713258/170427/NLF1Av6zZTQd4H7.mp4 (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x0000016A08D2EC50>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed',))




C:\Users\Hoogstra>anime dl "Tokyo Ghoul" --download-dir H:\AnimeDownload -q 1080p -ll DEBUG
INFO root: anime-downloader 3.6.0
DEBUG root: Platform: Windows-10-10.0.17134-SP0
DEBUG root: Python 3.6.4
ERROR root: [31mHTTPSConnectionPool(host='www4.9anime.is', port=443): Max retries exceeded with url: /search?keyword=Tokyo+Ghoul (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x000002418C062EF0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed',))[0m


```

**Comment by vn-ki** [comment]
9anime does not work.

For kissanime, can you try installing aria2c (follow instructions in wiki or place the exe in your path), and try again with the flag `-xd "{aria2}"`.

You can try other providers like twist.moe, masterani or animepahe.

**Comment by vn-ki** [comment]
I just released v3.6.1. You may try 9anime again.

**Comment by vn-ki** [comment]
I just released v3.6.1. You may try 9anime again.
